lib/v2/screens/profile_screens/profile/interactor/mappers/upgrade_citizenship_result_mapper.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:seeds/v2/domain-shared/page_command.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/v2/screens/profile_screens/profile/interactor/viewmodels/page_commands.dart';
+import 'package:seeds/v2/screens/profile_screens/profile/interactor/viewmodels/profile_state.dart';
+
+class UpgradeCitizenshipResultMapper extends StateMapper {
+  ProfileState mapResultToState(ProfileState currentState, Result result, bool isResident) {
+    String newCitizenshipStatus;
+
+    if (result.isError) {
+      // citizenship upgrade error, show snackbar fail
+      print('Error upgrading citizenship status');
+      return currentState.copyWith(
```

Comment:
Do we really need this print? Bloc lib already shows a lot info in the events logs.

Reply:
um I just want to show an error if the upgrade fails, give the user some feedback

Reply:
And you are doing it with the snack you fire, I was talking of the `print()`.

Reply:
oh the print()? I mean is good for debugging but sure I can remove it.

Reply:
done